src/contributors/SearchContributor.ts

```diff
@@ -0,0 +1,158 @@
+    public constructor(
+        identifier: string,
+        collaborativeSearcheService: CollaborativesearchService,
+        configService: ConfigService,
+        collection: string
+    ) {
+        super(identifier, configService, collaborativeSearcheService, collection);
+        this.collections = [];
+        this.collections.push({
+            collectionName: collection
+        });
+    }
+
+    public static getJsonSchema(): Object {
+        return jsonSchema;
+    }
+
+    public isUpdateEnabledOnOwnCollaboration() {
+        return false;
+    }
+
+    public fetchData(collaborationEvent: CollaborationEvent): Observable<[]> {
+        return from([]);
+    }
+
+    public computeData(data: { label: string; hits: Hits; }): { label: string; hits: Hits; } {
+        return data;
+    }
+    public setData(data: { label: string; hits: Hits; }): any {
+        return from([]);
+
+    }
+    public setSelection(collaboration: Collaboration): any {
+        return from([]);
+    }
+
+    /**
+    * @returns Pretty name of contributor.
+    */
+    public getFilterDisplayName(): string {
+        return '';
+    }
+    /**
+    * @returns Package name for the configuration service.
+    */
+    public getPackageName(): string {
+        return 'arlas.web.contributors.search';
+    }
+
+    public getAutoCompleteResponse$(search: string): Observable<AggregationResponse> {
+        const aggregation: Aggregation = {
+            type: Aggregation.TypeEnum.Term,
+            field: this.autoCompleteField,
+            include: search + '.*',
+            size: (this.autoCompleteSize).toString()
+        };
+        // Add filter to improve aggregation performances
+        const filterAgg: Filter = {
+            f: [[{
+                field: this.autoCompleteField,
+                op: Expression.OpEnum.Like,
+                value: search
+            }]]
+        };
+
+        this.searching = true;
+        return this.collaborativeSearcheService.resolveButNotAggregation(
+            [projType.aggregate, [aggregation]],
+            this.collaborativeSearcheService.collaborations,
+            this.collection,
+            this.identifier,
+            filterAgg
+        );
+
+    }
+
+
+    public search(value: string) {
+        if (value.trim() !== '') {
+            const filter: Filter = {
+                q: [[this.searchField + ':' + value.trim()]]
+            };
+            const collabFilters = new Map<string, Filter[]>();
+            collabFilters.set(this.collection, [filter]);
+            const collaboration: Collaboration = {
+                filters: collabFilters,
+                enabled: true
+            };
+
+            this.collaborativeSearcheService.setFilter(this.identifier, collaboration);
+        }
+    }
+
+    private getAutoCompleteSize(): number {
+        const autoCompleteSize = this.getConfigValue('autocomplete_size');
+        return (autoCompleteSize !== undefined) ? autoCompleteSize : 20;
```

Comment:
we could maybe have a const for the default autocomplete size, to make it easier to change/override if need be ?

Reply:
fixed